samples/app-cache-meetings/nodejs/src/components/app-cache-tab.tsx

```diff
@@ -0,0 +1,119 @@
+// <copyright file="app-cache-tab.tsx" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// </copyright>
+
+import { useState } from "react";
+import React from "react";
+import * as microsoftTeams from "@microsoft/teams-js";
+
+/// </summary>
+/// function logitem
+/// </summary>
+function logItem(action: string, actionColor: string, message: string) {
+    const newItem =
+        "<span style='font-weight:bold;color:" +
+        actionColor +
+        "'>" +
+        action +
+        "</span> " +
+        message +
+        "</br>";
+    return newItem;
+}
+
+/// </summary>
+/// In beforeUnloadHandler using setItems and readyToUnload callback function
+/// </summary>
+const beforeUnloadHandler = (
+    setItems: React.Dispatch<React.SetStateAction<string[]>>,
+    readyToUnload: () => void) => {
+
+    let newItem = logItem("OnBeforeUnload", "purple", "Started");
+    setItems((Items) => [...Items, newItem]);
+
+    // dispose resources and cleanup
+    newItem = logItem("OnBeforeUnload", "purple", "Completed");
+    setItems((Items) => [...Items, newItem]);
+
+    console.log("sending readyToUnload to TEAMS");
+    readyToUnload();
+
+    return true;
+};
+
+/// </summary>
+/// loadHandler using setItems to set values
+/// </summary>
+const loadHandler = (
+    setItems: React.Dispatch<React.SetStateAction<string[]>>,
+    data: microsoftTeams.LoadContext) => {
+    logItem("OnLoad", "blue", "Started for " + data.entityId);
+
```

Comment:
add a comment here "use the entityId, contentUrl to hydrate content before invoking notifySuccess" 
Started and Completed back to back could be confusing.

Reply:
Added comment, and also add "Route to specific page" between start and completed.